### PR TITLE
Add logarithmic and power projections

### PR DESCRIPTION
**Pull request by willschlitzer**
As discussed in #292, adding the Cartesian logarithmic and Cartesian power projections to the projection gallery.

### CODE REVIEW

File: examples/projections/nongeo/cartesian_power.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+"""
+Cartesian power
+===============
+
+``xwidth[p *p-value*]/[height [p *p-value*]]``: Give the ``width`` of the figure and
+the optional argument ``height``. The axis or axes with a logarithmic transformation
+requires ``"p"`` and the power transformation for that axis.
+"""
```
Comment (seisman) [inline examples/projections/nongeo/cartesian_power.py:8]: image

This part is difficult to read. We need to find a better way to show the syntax.

Comment (seisman) [reply examples/projections/nongeo/cartesian_power.py:8]: Related to this issue: https://github.com/GenericMappingTools/pygmt/issues/631

Comment (willschlitzer) [reply examples/projections/nongeo/cartesian_power.py:8]: Not sure of a  good way to do this; I'll have to give it some thought.

Comment (willschlitzer) [reply examples/projections/nongeo/cartesian_power.py]: @seisman Would you take a look at the updated docstring? I think it's in line with the suggestion on #631.

File: examples/projections/nongeo/cartesian_logarithmic.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# Create a list of x values for every 10 in 0-100
+xpoints = np.arange(0, 101, 10)
+# Create a list of y-values that are the square root of the x-values
+ypoints = xpoints ** 0.5
+
+fig = pygmt.Figure()
+fig.plot(
+    region=[1, 100, 0, 10],
+    # Set a logarithmic transformation on the x-axis
+    projection="x4cl/0.4c",
```
Comment (seisman) [inline examples/projections/nongeo/cartesian_logarithmic.py:25]: Any reason for lower-case `x` (map scale), not upper-case `X` (map width)?

Comment (willschlitzer) [reply examples/projections/nongeo/cartesian_logarithmic.py:25]: The example for the Cartesian power projection in the GMT docs used the lower case x, so I copied it from there. I'll try it with the capital x and change it to be consistent with the rest of the projection examples.

Comment (willschlitzer) [reply examples/projections/nongeo/cartesian_logarithmic.py]: I pushed a change using map width

File: examples/projections/nongeo/cartesian_power.py (hunk 1 above)
Comment (seisman) [reply examples/projections/nongeo/cartesian_power.py]: Let's keep this PR open until we reach an agreement in #631.

Comment (willschlitzer) [reply examples/projections/nongeo/cartesian_power.py]: Sounds good!

File: examples/projections/nongeo/cartesian_power.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+Cartesian power
+===============
+
+**x**\ *width*\ [**p**\ *p-value*]/[*height*\ [**p**\ *p-value*]]: Give the
```
Comment (seisman) [inline examples/projections/nongeo/cartesian_power.py:5]:
```suggestion
**X**\ *width*\ [**p**\ *pvalue*]/[*height*\ [**p**\ *pvalue*]]: Give the
```

File: examples/projections/nongeo/cartesian_power.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    # Set the line thickness to *thick*
+    # Use the default color *black* and the default style *solid*
+    pen="thick",
+    x=xvalues,
+    y=yvalues,
+)
+# Plot x,y values as points on the line
+# Style of points is 0.2 cm circles, color is *green* with a *black* outline
+# Points are set to clip if they go off the figure
+fig.plot(x=xvalues, y=yvalues, style="c0.2c", color="green", no_clip="c", pen="black")
```
Comment (seisman) [inline examples/projections/nongeo/cartesian_power.py:33]: What's the reason for using `no_clip` here?

File: examples/projections/nongeo/cartesian_logarithmic.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    frame=["WSne+gbisque", "x2g3", "ya2f1g2"],
+    x=xline,
+    y=yline,
+    # Set the line thickness to *1p*, the color to *blue*, and the style to *dash*
+    pen="1p,blue,-",
+)
+# Plot square root values as points on the line
+# Style of points is 0.3 cm square, color is *red* with a *black* outline
+# Points are set to clip if they go off the figure
+fig.plot(x=xpoints, y=ypoints, style="s0.3c", color="red", no_clip="c", pen="black")
```
Comment (seisman) [inline examples/projections/nongeo/cartesian_logarithmic.py:36]: What's the reason for using no_clip here?

File: examples/projections/nongeo/cartesian_power.py (hunk 4 above)
Comment (willschlitzer) [reply examples/projections/nongeo/cartesian_power.py:33]: It was in the example in the GMT docs, and I was trying to replicate them.

Comment (seisman) [reply examples/projections/nongeo/cartesian_power.py:33]: https://docs.generic-mapping-tools.org/dev/cookbook/coordinate-transformations.html#cartesian-logarithmic-projection

Do you mean this example?

Comment (willschlitzer) [reply examples/projections/nongeo/cartesian_power.py:33]: Yes

File: examples/projections/nongeo/cartesian_power.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
```
Comment (seisman) [inline examples/projections/nongeo/cartesian_power.py:33]:
```suggestion
fig.plot(x=xvalues, y=yvalues, style="c0.2c", color="green", no_clip=True, pen="black")
```

File: examples/projections/nongeo/cartesian_logarithmic.py (hunk 5 above)
Comment (seisman) [inline examples/projections/nongeo/cartesian_logarithmic.py:36]:
```suggestion
fig.plot(x=xpoints, y=ypoints, style="s0.3c", color="red", no_clip=True, pen="black")
```

**Review by seisman — COMMENTED** [review]
Let's use `no_clip=True` for both examples. 

The examples don't have to be the same as in the GMT cookbook.

File: examples/projections/nongeo/cartesian_logarithmic.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    # Set the figures frame, color, and gridlines
+    frame=["WSne+gbisque", "x2g3", "ya2f1g2"],
+    x=xline,
+    y=yline,
+    # Set the line thickness to *1p*, the color to *blue*, and the style to *dash*
+    pen="1p,blue,-",
+)
+# Plot square root values as points on the line
+# Style of points is 0.3 cm square, color is *red* with a *black* outline
+# Points are set to clip if they go off the figure
```
Comment (seisman) [inline examples/projections/nongeo/cartesian_logarithmic.py:35]:
```suggestion
# Points are not clipped if they go off the figure
```

File: examples/projections/nongeo/cartesian_power.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    frame=["WSne+givory", "xa1p", "ya2f1"],
+    # Set the line thickness to *thick*
+    # Use the default color *black* and the default style *solid*
+    pen="thick",
+    x=xvalues,
+    y=yvalues,
+)
+# Plot x,y values as points on the line
+# Style of points is 0.2 cm circles, color is *green* with a *black* outline
+# Points are set to clip if they go off the figure
```
Comment (seisman) [inline examples/projections/nongeo/cartesian_power.py:32]:
```suggestion
# Points are not clipped if they go off the figure
```

**Pull request merged**